bundles/org.openhab.core.transform/src/main/java/org/openhab/core/transform/FileTransformationProvider.java

```diff
@@ -75,7 +75,7 @@ public FileTransformationProvider(
         watchService.registerListener(this, transformationPath);
         // read initial contents
         try {
-            Files.walk(transformationPath).filter(Files::isRegularFile)
+            Files.walk(transformationPath).filter(Files::isRegularFile).map(transformationPath::relativize)
```

Comment:
Can you also use try-with-resources with it?

See the JavaDocs for `Files.walk`

```java
     * This method must be used within a try-with-resources statement or similar
     * control structure to ensure that the stream's open directories are closed
     * promptly after the stream's operations have completed.
```

It might also be an issue in other places where `Files.walk` is used and the stream is not closed.

Reply:
Done. I have also improved the tests a little bit to make them more realistic (the mocked `WatchService` is now used in the same way as it is in the real-world application).